benchexec/tracerx.py

```diff
@@ -0,0 +1,111 @@
+import benchexec.tools.template
+import benchexec.model
+
+
+class Tool(benchexec.tools.template.BaseTool):
+    """
+    Tool info for Tracer-X (https://www.comp.nus.edu.sg/~tracerx/).
+    """
+
+    REQUIRED_PATHS = ["bin", "include", "klee_build", "libraries"]
+
+    def executable(self):
+        return util.find_executable("bin/tracerx")
+
+    def program_files(self, executable):
+        return self._program_files_from_executable(
+            executable, self.REQUIRED_PATHS, parent_dir=True
+        )
+
+    def version(self, executable):
+        """
+        The output looks like this:
+        Tracer-X 0.2.0 (https://www.comp.nus.edu.sg/~tracerx/)
+          Built August 19 2019
+          Build mode: Release
+          Build revision: 60794e3eac58744548657500e425241b57f4bdb7
+        LLVM (http://llvm.org/):
+          LLVM version 3.4.2
+          Optimized build.
+          Built Oct 15 2014 (13:57:47).
+          Default target: x86_64-pc-linux-gnu
+          Host CPU: bdver1
+        """
+        stdout = self._version_from_tool(executable)
+        line = next(l for l in stdout.splitlines() if l.startswith("KLEE"))
+        line = line.replace("KLEE", "")
+        line = line.split("(")[0]
+        return line.strip()
+
+    def cmdline(self, executable, options, tasks, propertyfile=None, rlimits={}):
+        if propertyfile:
+            options += ["--property-file=" + propertyfile]
+        if benchexec.model.MEMLIMIT in rlimits:
+            options += ["--max-memory=" + str(rlimits[benchexec.model.MEMLIMIT])]
+        if benchexec.model.TIMELIMIT in rlimits:
+            options += ["--max-time=" + str(rlimits[benchexec.model.TIMELIMIT])]
+        if benchexec.model.WALLTIMELIMIT in rlimits:
+            options += ["--max-walltime=" + str(rlimits[benchexec.model.WALLTIMELIMIT])]
+        if benchexec.model.SOFTTIMELIMIT in rlimits:
+            options += [
+                "--max-cputime-soft=" + str(rlimits[benchexec.model.SOFTTIMELIMIT])
+            ]
+        if benchexec.model.HARDTIMELIMIT in rlimits:
+            options += [
+                "--max-cputime-hard=" + str(rlimits[benchexec.model.HARDTIMELIMIT])
+            ]
+
+        return [executable] + options + tasks
+
+    def name(self):
+        return "Tracer-X"
+
+    def determine_result(self, returncode, returnsignal, output, isTimeout):
+        """
+        Parse the output of the tool and extract the verification result.
+        This method always needs to be overridden.
+        If the tool gave a result, this method needs to return one of the
+        benchexec.result.RESULT_* strings.
+        Otherwise an arbitrary string can be returned that will be shown to the user
+        and should give some indication of the failure reason
+        (e.g., "CRASH", "OUT_OF_MEMORY", etc.).
+        """
+        for line in output:
+            if line.startswith("KLEE: ERROR: "):
+                if line.find("ASSERTION FAIL:") != -1:
```

Comment:
Instead of all the calls to `find`, just do `if "foo" in line`.

Reply:
fixed.